rye/src/cli/init.rs

```diff
@@ -282,3 +353,205 @@ pub fn execute(cmd: Args) -> Result<(), Error> {
 
     Ok(())
 }
+
+#[derive(Default)]
+struct Metadata {
+    name: String,
+    version: String,
+    description: String,
+    author: Option<(String, String)>,
+    requires_python: Option<String>,
+    license: Option<String>,
+    dependencies: Option<Vec<String>>,
+    dev_dependencies: Option<Vec<String>>,
+}
+
+/// Check if setup.py or setup.cfg exist.
+fn import_available<T: AsRef<Path>>(dir: T) -> bool {
+    let dir = dir.as_ref();
+    dir.join("setup.py").is_file() | dir.join("setup.cfg").is_file()
```

Comment:
I'm not a fan of `|` instead of `||`. Same for the other one.

Reply:
Oh cool `||` is a lazy operator. TIL

Reply:
083de3442807e26ccb1176f0b2501e0419b4eb59